src/rest_api/class-openagenda-controller.php

```diff
@@ -725,11 +725,92 @@ public function submit_item( $request ) {
 		// Get custom fields via CMB2 REST API.
 		$cmb2_boxes = \CMB2_Boxes::get_all();
 
-		if ( ! $cmb2_boxes ) {
-			return new \WP_Error( 'rest_missing_fields', __( 'Can\'t submit event, because CMB2 custom fields are missing.', 'openagenda-base' ), array( 'status' => 400 ) );
+		// Check which cmb2 fields are required and are not submitted in the request or are empty.
+		$event_prefix    = 'event_';
+		$required_fields = array();
+		foreach ( $cmb2_boxes as $cmb2_box ) {
+			$cmb2_box = $cmb2_box->__get( 'meta_box' );
+			if ( in_array( 'event', $cmb2_box['object_types'], true ) ) {
+				$cmb2_box_fields = $cmb2_box['fields'];
```

Comment:
Hoewel je hier op de volgende regel checkt of er wel iets in `$cmb2_box_fields` zit, zou je toch moeten controleren of `$cmb2_box['fields']` wel bestaat.